python/ql/test/experimental/dataflow/model-summaries/model_summaries.py

```diff
@@ -0,0 +1,128 @@
+
+import sys
+import os
+
+sys.path.append(os.path.dirname(os.path.dirname((__file__))))
+from testlib import expects
+
+# These are defined so that we can evaluate the test code.
+NONSOURCE = "not a source"
+SOURCE = "source"
+
+
+def is_source(x):
+    return x == "source" or x == b"source" or x == 42 or x == 42.0 or x == 42j
+
+
+def SINK(x):
+    if is_source(x):
+        print("OK")
+    else:
+        print("Unexpected flow", x)
+
+
+def SINK_F(x):
+    if is_source(x):
+        print("Unexpected flow", x)
+    else:
+        print("OK")
+
+ensure_tainted = ensure_not_tainted = print
+TAINTED_STRING = "TAINTED_STRING"
+
+from foo import MS_identity, MS_apply_lambda, MS_reversed, MS_list_map, MS_append_to_list
+
+# Simple summary
+via_identity = MS_identity(SOURCE)
+SINK(via_identity)  # $ flow="SOURCE, l:-1 -> via_identity"
+
+tainted = MS_identity(TAINTED_STRING)
+ensure_tainted(tainted) # $ tainted
```

Comment:
Since you have just shown that we have data-flow, and we know all dataflow steps are also taint-flow steps, I think it's fine to only check taint-flow for the cases where there is NOT dataflow.

I think that will make the test-file a bit easier to read as well.

Reply:
Good point. I tried to make it more consistent now by not checking taint when dataflow is already established and when we do check taint, check both the collection and the expected element.

Reply:
Thinking some more about this, I think these tests should be about whether you can write flow summaries in CSV files that do the right thing.

Having all these extra taint steps are closer to what our current taint-tracking does in the end, but from my point of view, doesn't help us achieve the goal of the tests.

I removed the `TAINTED_LIST` parts locally, but thought it was a bit too controversial to just commit directly to your PR -- instead I've put the commits here: https://github.com/yoff/codeql/pull/79 (if you agree, we can add these to main later on :+1:)